security-setup

```diff
@@ -415,27 +428,30 @@ class Mesos(Component):  # Mesos should always come after any frameworks
 
     def framework_auth(self):
         "framework auth"
-        with self.modify_security() as config:
-            if 'marathon_principal' in config and 'marathon_secret' in config:
+        if not self.args.no_authenticate_frameworks:
+            with self.modify_security() as config:
                 config.setdefault('mesos_credentials', [])
-                credential = {
-                    'principal': config['marathon_principal'],
-                    'secret': config['marathon_secret'],
-                }
-                if credential not in config['mesos_credentials']:
-                    config['mesos_credentials'].append(credential)
-                    print('set auth for Marathon')
+                if 'marathon_principal' in config and 'marathon_secret' in config:
```

Comment:
this is implicit by the class ordering - see the comment on the class definition line. If we're checking for the credentials to be set, however, it needs to *not* fail silently. Can you add a warning, please?